src/org/openstreetmap/josm/data/osm/Way.java

```diff
@@ -44,6 +44,11 @@ public List<Node> getNodes() {
         return new CopyList<>(nodes);
     }
 
+    @Override
+    public List<Node> getChildren() {
+        return getNodes();
+    }
+
```

Comment:
See comment in IPrimitive. It should either be there or IWay. Putting it in IWay is easy; `public List<N> getChildren() { return getNodes(); }`